### DISCUSSION TITLE
Set a system variable in a hop run execution inside container

### DISCUSSION THREAD

**Discussion by openinbigdata** [apache/hop#4519] [General]
Hi Guys,

I need to change the value of the HOP_LOG_MARK_MAPPINGS system variable to use it with hop run via docker. Via Hop GUI, this was easy, now via hop run and inside a container I'm not able to define this variable. Does anyone have any tips on how to do this?

Thanks,

**Comment by bamaer** [comment]
I guess you'll have to extend the default image to support it. Feel free to create a ticket to have it included in the default image.

**Comment by dave-csc** [comment]
You could try to pass the system variable as a (Docker) environment variable, e.g.:
`docker run -it --env HOP_SYSTEM_PROPERTIES=HOP_LOG_MARK_MAPPINGS=Y [...]`

Hope this helps

**Comment by florekk925** [comment]
Hello

I tried this tip but It does not work (in my configuration?).
I added the appropriate entries to the k8s configuration.
                - name: HOP_SYSTEM_PROPERTIES
                  value: "HOP_LOG_MARK_MAPPINGS=Y"
In the log I have an entry:
2025/02/16 20:00:05 - Setting system properties at runtime: HOP_LOG_MARK_MAPPINGS=Y

But in the workflow logs there are no complete path.
Where is the problem?

**Reply by hansva** [reply]
that variable can't be changed with an environment variable and requires patching the hop-config.json file.
